net/bonding/files/lib/netifd/proto/bonding.sh

```diff
@@ -0,0 +1,237 @@
+			[ -n "$primary_reselect" ] && echo "$primary_reselect" > /sys/class/net/"$link"/bonding/primary_reselect
+			[ -n "$fail_over_mac" ] && echo "$fail_over_mac" > /sys/class/net/"$link"/bonding/fail_over_mac
+			[ -n "$num_grat_arp__num_unsol_na" ] && echo "$num_grat_arp__num_unsol_na" > /sys/class/net/"$link"/bonding/num_grat_arp
+			[ -n "$num_grat_arp__num_unsol_na" ] && echo "$num_grat_arp__num_unsol_na" > /sys/class/net/"$link"/bonding/num_unsol_na
+			[ -n "$xmit_hash_policy" ] && echo "$xmit_hash_policy" > /sys/class/net/"$link"/bonding/xmit_hash_policy
+		;;
+        esac
+
+	# Set link monitoring (with corresponding parameters)
+	local link_monitoring
+	json_get_vars link_monitoring
+
+	case "$link_monitoring" in
+
+		arp)
+			local arp_interval arp_ip_target arp_all_targets arp_validate
+			json_get_vars arp_interval arp_ip_target arp_all_targets arp_validate
+
+			[ -n "$arp_interval" -a "$arp_interval" != 0 ] && echo "$arp_interval" > /sys/class/net/"$link"/bonding/arp_interval
+
+			IFS=' '
+			for target in $arp_ip_target; do
+				echo "+$target" > /sys/class/net/"$link"/bonding/arp_ip_target
+			done
+
+			[ -n "$arp_all_targets" ] && echo "$arp_all_targets" > /sys/class/net/"$link"/bonding/arp_all_targets
+			[ -n "$arp_validate" ] && echo "$arp_validate" > /sys/class/net/"$link"/bonding/arp_validate
+		;;
+
+		mii)
+			local miimon downdelay updelay use_carrier
+			json_get_vars miimon downdelay updelay use_carrier
+
+			[ -n "$miimon" -a "$miimon" != 0 ] && echo "$miimon" > /sys/class/net/"$link"/bonding/miimon
+			[ -n "$downdelay" ] && echo "$downdelay" > /sys/class/net/"$link"/bonding/downdelay
+			[ -n "$updelay" ] && echo "$updelay" > /sys/class/net/"$link"/bonding/updelay
+			[ -n "$use_carrier" ] && echo "$use_carrier" > /sys/class/net/"$link"/bonding/use_carrier
+		;;
+	esac
+
+	# Add slaves to bonding interface
+	local slaves
+	json_get_vars slaves
+
+	for slave in $slaves; do
+
+		if [ "$(cat /proc/net/dev |grep "$slave")" == "" ]; then
+			echo "$cfg" "ERROR IN CONFIGURATION - $slave: No such device"
+			proto_notify_error "$cfg" "ERROR IN CONFIGURATION - $slave: No such device"
+			proto_block_restart "$cfg"
+			return
+		fi
+
+		ifconfig "$slave" down
+
+		sleep 1
+
+		echo "+$slave" > /sys/class/net/"$link"/bonding/slaves
+
+		ifconfig "$slave" up
+	done
+
+	[ -n "$all_slaves_active" ] && echo "$all_slaves_active" > /sys/class/net/"$link"/bonding/all_slaves_active
+
+	# Check for static or DHCP configuration
+	local dhcp
+	json_get_vars dhcp
+
+	local ipaddr netmask
+	json_get_vars ipaddr netmask
+
+	# ATTENTION
+	#All json vars have to be read before the line below, as the
+	# json object will be overwritten by proto_init_update
+	# ATTENTION
+
+	proto_init_update "$link" 1
+
+	# For static configuration we _MUST_ have an IP address
+	[ -z "$ipaddr" ] && {
+		echo "$cfg" "MISSING STATIC IP ADDRESS"
+		proto_notify_error "$cfg" "MISSING STATIC IP ADDRESS"
```

Comment:
Please use one of the existing error codes, e.g. `MISSING_ADDRESS` (used by vxlan, vti, 6in4) or `INVALID_LOCAL_ADDRESS` (used by 6to4).

Reply:
OK